assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -1294,6 +1295,34 @@ public SimpleSubmission(AssignmentSubmission as, boolean anonymousGrading) {
                     this.late = dateSubmitted.compareTo(as.getAssignment().getDueDate()) > 0;
                 }
                 this.submittedAttachments = as.getAttachments();
+
+                SecurityAdvisor securityAdvisor = (String userId, String function, String reference) -> {
+
+                    if (ContentHostingService.AUTH_RESOURCE_READ.equals(function)) {
+                        return SecurityAdvisor.SecurityAdvice.ALLOWED;
+                    } else {
+                        return SecurityAdvisor.SecurityAdvice.NOT_ALLOWED;
+                    }
+                };
+
+                try {
+                    securityService.pushAdvisor(securityAdvisor);
+                    this.submittedAttachments.forEach(ref -> {
+
+                        try {
+                            ResourceProperties props = contentHostingService.getProperties(ref.replaceFirst("\\/content", ""));
+                            String previewId = props.getProperty(ContentHostingService.PREVIEW);
+                            System.out.println(previewId);
+                            if (StringUtils.isNotEmpty(previewId)) {
+                                previewableAttachments.put(ref, contentHostingService.getResource(previewId).getReference());
+                            }
+                        } catch (Exception e) {
+                            e.printStackTrace();
```

Comment:
log.warn("Could not access properties for resource {}, {}", resourceId, e.toString())
 message vs e.printStackTrace